### PR TITLE
Fix/other tracestate missing

### PR DESCRIPTION
**Pull request by vgenev**
Fixed losing of other vendors tracestate

### CODE REVIEW

File: src/Span.ts [hunk 1]
```diff
@@ -226,6 +231,42 @@ class Span implements Partial<ISpan> {
...
+
+  setTracestateTags(tags: TraceTags): this {
+    let stateHashMap = { _rest: '' }
+    if (!!this.spanContext.tags && !!this.spanContext.tags.tracestate) {
+      stateHashMap = Util.getOwnTracestateMap(Config.EVENT_LOGGER_VENDOR_PREFIX, this.spanContext.tags.tracestate)
+    }
+    const rest = stateHashMap._rest
+    delete stateHashMap._rest
+    const tracestateToSet = { ...stateHashMap, ...tags }
+    // const newContext = {...this.spanContext, ...{ tags: {...this.spanContext.tags, ...{ tracestate: `${Config.EVENT_LOGGER_VENDOR_PREFIX}=${Util.hashMapToString(tracestateToSet)},${rest}` } } } }
```
Comment (yosheeck) [inline src/Span.ts]: 2 lines of comments to remove I guess ?

File: src/lib/util.ts [hunk 2]
```diff
@@ -65,16 +65,46 @@ function shouldOverrideEvent(overrideDict: { [index: string]: boolean }, eventTy
   return false
 }
 
+function hashMapToString (obj: { [key: string] : string }): string {
```
Comment (mdebarros) [inline src/lib/util.ts]: Please either use `reduce` or lodash to do this functionality.

File: src/lib/util.ts [hunk 3]
```diff
@@ -65,16 +65,46 @@ function shouldOverrideEvent(overrideDict: { [index: string]: boolean }, eventTy
...
+} 
+
+const getOwnTracestateMap = (vendor: string, tracestate: string): { [key: string]: string, _rest: string } => {
+  if (!tracestate.includes(vendor)) {
+    return { _rest: tracestate }
+  }
+  const stateHashMap: { [key: string]: string, _rest: string } = { _rest: '' }
+  const t = tracestate
+    .split(',') // get each vendor
+    .filter(ts => ts.split('=')[0] === vendor)[0] //get own vendor
```
Comment (mdebarros) [inline src/lib/util.ts]: Rather check the key of the key-value to make sure its the "spanId".

Comment (vgenev) [reply src/lib/util.ts]: this is getting our tags as hash map and the rest as the string it is, yet.
so what the check would give us exactly? just to make sure we have it?

**Pull request merged**